CMakeLists.txt

```diff
@@ -0,0 +1,256 @@
+cmake_minimum_required(VERSION 3.15.0)
+project(mapnik 
+    VERSION 3.0.24 
+    HOMEPAGE_URL "https://mapnik.org/" 
+    DESCRIPTION "Mapnik is an open source toolkit for developing mapping applications" 
+    LANGUAGES CXX 
+)
+
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+
+option(BUILD_SHARED_LIBS "build mapnik dynamic" ON)
+option(BUILD_TEST "builds the tests" ON)
+option(USE_EXTERNAL_MAPBOX_GEOMETRY "Use a external mapnik/geometry.hpp. If off, use the submodule" OFF)
+option(USE_EXTERNAL_MAPBOX_POLYLABEL "Use a external mapnik/polylabel. If off, use the submodule" OFF)
+option(USE_EXTERNAL_MAPBOX_PROTOZERO "Use a external mapnik/protozero. If off, use the submodule" OFF)
+option(USE_EXTERNAL_MAPBOX_VARIANT "Use a external mapnik/variant. If off, use the submodule" OFF)
+option(USE_JPEG "adds jpeg support" ON)
+option(USE_PNG "adds png support" ON)
+option(USE_TIFF "adds tiff support" ON)
+option(USE_WEBP "adds webp support" ON)
+option(USE_LIBXML2 "adds libxml2 support" ON)
+option(USE_CAIRO "adds the cairo renderer" ON)
+option(USE_PROJ4 "adds proj4 support" ON)
+option(USE_GRID_RENDERER "adds grid renderer" ON)
+option(USE_SVG_RENDERER "adds grid renderer" ON)
+
+option(USE_PLUGIN_INPUT_CSV "adds plugin input csv" ON)
+option(USE_PLUGIN_INPUT_GDAL "adds plugin input gdal" ON)
+option(USE_PLUGIN_INPUT_GEOBUF "adds plugin input geobuf" ON)
+option(USE_PLUGIN_INPUT_GEOJSON "adds plugin input geojson" ON)
+option(USE_PLUGIN_INPUT_OGR "adds plugin input ogr" ON)
+option(USE_PLUGIN_INPUT_PGRASTER "adds plugin input pgraster" ON)
+option(USE_PLUGIN_INPUT_POSTGIS "adds plugin input postgis" ON)
+option(USE_PLUGIN_INPUT_RASTER "adds plugin input raster" ON)
+option(USE_PLUGIN_INPUT_SHAPE "adds plugin input shape" ON)
+option(USE_PLUGIN_INPUT_SQLITE "adds plugin input sqlite" ON)
+option(USE_PLUGIN_INPUT_TOPOJSON "adds plugin input topojson" ON)
+
+option(BUILD_DEMO_VIEWER "builds the demo viewer" ON)
+option(BUILD_DEMO_CPP "builds the demo c++ application" ON)
+
+option(BUILD_UTILITY_GEOMETRY_TO_WKB "builds the utility program geometry_to_wkb" ON)
+option(BUILD_UTILITY_MAPNIK_INDEX "builds the utility program mapnik_index" ON)
+option(BUILD_UTILITY_MAPNIK_RENDER "builds the utility program mapnik_render" ON)
+option(BUILD_UTILITY_OGRINDEX "builds the utility program ogrindex" OFF)
+option(BUILD_UTILITY_PGSQL2SQLITE "builds the utility program pgsql2sqlite" ON)
+option(BUILD_UTILITY_SHAPEINDEX "builds the utility program shapeindex" ON)
+option(BUILD_UTILITY_SVG2PNG "builds the utility program svg2png" ON)
+
+set(PLUGINS_INSTALL_DIR bin/plugins/input CACHE STRING "installs the plugins in the specified directory")
```

Comment:
Mapnik installs plugins by default to `bin/mapnik/input`, see https://github.com/mapnik/mapnik/blob/4227fd5d698d069a92917cfea9ce9139435d3f12/SConstruct#L1393

Reply:
Thanks for the hint!
Will change to the default path.

Reply:
`lib/mapnik/input` @mathisloge ^

Reply:
(but `bin/mapnik/input` in Windows)

Reply:
:+1:  @SNiLD @mathisloge yep, I don't use Windows often :)

Reply:
I will add a default value for Linux and windows. In the end it should be almost irrelevant, because the generated MapnikTargets.cmake have a variable with the plugin and font dir. And with that you can either copy the plugins with your project cmake or configure a file to include the file. Like the viewer example viewer.ini

Reply:
Or you could set the variable for the installation directory as you wish.

But für windows the default value will be /bin/mapnik/input and for Linux it will be /lib/mapnik/input